### DISCUSSION TITLE
How to get data for longer duration simulations?

### DISCUSSION THREAD

**Discussion by Ramesh-Isukapalli** [pybamm-team/PyBaMM#3547] [Q&A]
I'm trying to run the simulation for longer periods of time like about 3000 cycles or 80% termination capacity, however, My 12GB RAM has been consumed well before 2000 cycles reached. I want to do modelling for RUL using 80% capacity as threshold for that I need to have the data for almost 3000 cycles for 30 cells. How can I optimize my RAM. Can someone please advise or suggest any ideas?

**Comment by valentinsulzer** [comment]
Check out https://docs.pybamm.org/en/latest/source/examples/notebooks/simulating-long-experiments.html - the `save_at_cycles` argument should be useful for yo

**Reply by Ramesh-Isukapalli** [reply]
Hi @tinosuzler,

Thanks for the reply, I want to have the data at every cycle to analyze. How should I workout on that part?
